test/podman-tests.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Do not set -e as we want to work on all results of `rpm -q dnf[5]`.
+set -xo pipefail
+
+if [[ $1 == '' ]]; then
+    echo -e "Usage: podman-tests.sh TEST_TYPE STREAM\nTEST_TYPE can be 'e2e' or 'system'\nSTREAM can be 'upstream' or 'downstream'"
+    exit 0
+fi
+
+# `rhel` macro exists on RHEL, CentOS Stream, and Fedora ELN
+# `centos` macro exists only on CentOS Stream
+CENTOS_VERSION=$(rpm --eval '%{?centos}')
+RHEL_VERSION=$(rpm --eval '%{?rhel}')
+
+# For upstream tests, we need to test with podman and other packages from the
+# podman-next copr. For downstream tests (bodhi, errata), we don't need any
+# additional setup
+if [[ $2 == "upstream" ]]; then
+    # Use CentOS Stream 10 copr target for RHEL-10 until EPEL 10 becomes
+    # available
+    if [[ -n $CENTOS_VERSION || $RHEL_VERSION -ge 10 ]]; then
+        dnf -y copr enable rhcontainerbot/podman-next centos-stream-$CENTOS_VERSION
+    else
+        dnf -y copr enable rhcontainerbot/podman-next
+    fi
```

Comment:
These you can add in the `plan` section
```yaml
prepare:
    how: install
    copr: rhcontainerbot/podman-next
    package: podman
```

Adding multiple copr, I am not sure, but you can try to add a similar `prepare` step with empty `package`.

Reply:
That doesn't work if i want to enable centos-stream copr targets on centos-stream envs. All EL envs end up choosing epel by default.

Sure, we could do multiple prepare steps, but then we end up with too many prepare steps and there's concern that we may miss some distro-version-arch combo. It would probably be much easier if a single `adjust` could handle if-else in addition to a `default & exception` way.

Reply:
> It would probably be much easier if a single adjust could handle if-else in addition to a default & exception way.

Hmm, good point. I don't quite have a clean solution for this. The only thing I can think of is to have `fedora` (or a blank common) as a default (`else` variant) and use `prepare-:` to delete steps, and `prepare.[].order` to get a specific ordering.

But of course there is also the re-usability of the scripts, so it might not be relevant rn anyway.